application-analytics-ui/src/main/resources/Analytics/Code/Macros/MostViewedPages.xml

```diff
@@ -291,7 +291,11 @@
 #end
 #if ($services.licensing.licensor.hasLicensureForEntity(
   $services.model.createDocumentReference('', ['Analytics', 'Code'], 'ConfigurationClass')))
-  #executeMacro()
+  #if ($xwiki.hasAccessLevel('view', 'Analytics.WebHome'))
+    #executeMacro()
+  #else
+    {{error}}$escapetool.xml($services.localization.render('analytics.accessDenied')){{/error}}
```

Comment:
I don't see any HTML macro, so I'm not sure why you escape XML special characters.

```suggestion
    {{error}}{{translation key="analytics.accessDenied" /}}{{/error}}
```